public/components/notebook/notebook_name_modal.tsx

```diff
@@ -16,22 +16,54 @@ import {
   EuiModalHeaderTitle,
 } from '@elastic/eui';
 import React, { useState, useCallback } from 'react';
+import { useCore } from '../../contexts/core_context';
+import { toMountPoint } from '../../../../../src/plugins/opensearch_dashboards_react/public';
 
-interface Props {
+export interface NotebookNameModalProps {
   onClose: () => void;
   // SaveChat hook depends on context. Runtime modal component can't get context, so saveChat needs to be passed in.
   saveChat: (name: string) => void;
 }
 
-export const NotebookNameModal = ({ onClose, saveChat }: Props) => {
+export const NotebookNameModal = ({ onClose, saveChat }: NotebookNameModalProps) => {
+  const {
+    services: {
+      notifications: { toasts },
+    },
+  } = useCore();
   const [name, setName] = useState('');
   const [loading, setLoading] = useState(false);
 
   const onSubmit = useCallback(async () => {
     setLoading(true);
-    await saveChat(name);
+    try {
+      const notebookId = await saveChat(name);
```

Comment:
The type of saveChat is `(name: string) => void;`, but here it expects it to be a promise which returns a `Promise<string>`.